File: solutions/_1624_largest_substring_between_two_equal_characters/solution.py
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        max_len = -1
        char_to_index = {}
        for i, char in enumerate(s):
            if char in char_to_index:
                max_len = max(max_len, i - char_to_index[char] - 1)
            else:
                char_to_index[char] = i
        return max_len


# https://leetcode.com/problems/largest-substring-between-two-equal-characters/

File: solutions/_1624_largest_substring_between_two_equal_characters/solution.java
package _1624_largest_substring_between_two_equal_characters;

import java.util.*;

class Solution {
    public int maxLengthBetweenEqualCharacters(String s) {
        Map<Character, Integer> map = new HashMap<>();
        int max = -1;
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (map.containsKey(c)) {
                max = Math.max(max, i - map.get(c) - 1);
            } else {
                map.put(c, i);
            }
        }
        return max;
    }
}

/**
 * https://leetcode.com/problems/largest-substring-between-two-equal-characters/
 */